code/modules/antagonists/traitor/contractor/items/contractor_uplink.dm

```diff
@@ -0,0 +1,42 @@
+/**
+  * # Contractor Uplink
+  *
+  * A contractor's point of contact with their Contractor Hub.
+  */
+/obj/item/contractor_uplink
+	name = "contractor uplink"
+	desc = "A standard, Syndicate issued tablet for handling important contracts while on the field."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "contractor_uplink"
+	w_class = WEIGHT_CLASS_SMALL
+	slot_flags = SLOT_BELT
+	origin_tech = "programming=5;syndicate=4" // Hackerman encryption
+	/// The Contractor Hub associated with this uplink.
+	var/datum/contractor_hub/hub = null
+
+/obj/item/contractor_uplink/Destroy()
+	// Right now, one uplink = one hub so this is fine.
+	QDEL_NULL(hub)
+	return ..()
+
+/obj/item/contractor_uplink/attack_self(mob/user)
+	hub.tgui_interact(user)
+
+/**
+  * Sends a message to the mob holding this item.
+  *
+  * Arguments:
+  * * text - The text to send.
+  * * snd - The sound to play to the holder only.
+  */
+/obj/item/contractor_uplink/proc/message_holder(text, snd)
+	var/mob/living/M = loc
+	while(!istype(M) && M?.loc)
+		M = M.loc
+	if(!istype(M))
+		return
+
+	to_chat(M, "<span class='notice'>[bicon(src)] Incoming encrypted transmission from your handlers. Message as follows:</span><br />"\
+			 + "<span class='boldnotice'>[text]</span>")
+	if(snd)
```

Comment:
`snd`? Bit of an odd name for `play_sound` or the like

Reply:
Renamed to `sndfile`